### ISSUE TITLE
Is `serial` the best data type for the `id` field in new tables?

### ISSUE THREAD

**Issue by mhkeller** [TablePlus/TablePlus#3437]
### Please answer the questions below, it helps us to keep track of the issue. Please do not ignore it or your issue will be closed.

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):

Postgres 16

3. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):

Build 496

5. The steps to reproduce this issue:

Create a new table and the default fields will be `id` with a type of `serial`

![Screenshot 2024-11-12 at 12 56 09](https://github.com/user-attachments/assets/036d2b20-7edb-4b57-a037-d76eea5b1457)

## The issue

In reading the Postgres ["Don't Do This" Wiki](https://wiki.postgresql.org/wiki/Don%27t_Do_This), it says to [not use](https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_serial) the `serial` type and instead use [Identity Columns](https://www.postgresql.org/docs/current/ddl-identity-columns.html). 

I may be a good idea to change the default settings here to be a `bigint` with a `GENERATED ALWAYS AS IDENTITY` clause, per the Postgres docs.

**Comment by huyphams** [comment]
Hi @mhkeller, int is a 4-byte data type, and bigint is an 8-byte data type. It's up to the user to choose, so if you're unsure of your table's purpose, it doesn't matter whether you use int or bigint.

**Comment by mhkeller** [comment]
My comment wasn't really about `int` vs `bigint`. Rather, if look at that wiki link, it says to not use the `serial` type and instead use Identity Columns. After reading https://github.com/TablePlus/TablePlus/issues/3410, it seems that there could be ways to make this easier by adding support either via defaults, dropdown options or both.

**Comment by bodinsamuel** [comment]
Agreed, it would great if TablePlus adapt the default key to integer identity when PG >= 10. This is the recommended way.
